src/log_format_impls.cc

```diff
@@ -146,7 +146,7 @@ class generic_log_format : public log_format {
                 &level)) != NULL) {
             const char *level_str = &sbr.get_data()[level.c_begin];
             logline::level_t level_val = logline::string2level(
-                    level_str, level.length());
+                    level_str, strlen(level_str));
```

Comment:
I don't think level_str is guaranteed to be null-terminated anymore.  The call to level.length() returns the length of the capture, which should be what you want.  The log formats returned by get_pcre_log_formats determines what part of the lines is used for the level.  The first capture in the regex is the timestamp and the second is the level.